plugins/webp-uploads/hooks.php

```diff
@@ -803,3 +803,69 @@ function webp_uploads_enable_additional_mime_type_support_for_all_sizes( array $
 	return $allowed_sizes;
 }
 add_filter( 'webp_uploads_image_sizes_with_additional_mime_type_support', 'webp_uploads_enable_additional_mime_type_support_for_all_sizes' );
+
+/**
+ * Converts palette PNG images to true color PNG images.
+ *
+ * GD cannot convert palette-based PNG to WebP/AVIF formats, causing conversion failures.
+ * This function detects and converts palette PNG to truecolor during upload.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<string, mixed>|mixed $file The uploaded file data.
+ * @return array<string, mixed> The modified file data.
+ */
+function webp_uploads_convert_palette_png_to_truecolor( $file ): array {
+	if ( ! is_array( $file ) ) {
+		$file = array();
+	}
+	if ( ! isset( $file['tmp_name'], $file['name'] ) ) {
+		return $file;
+	}
+	if ( isset( $file['type'] ) && is_string( $file['type'] ) ) {
+		if ( 'image/png' !== strtolower( $file['type'] ) ) {
+			return $file;
+		}
+	} elseif ( 'image/png' !== wp_check_filetype( $file['name'] )['type'] ) {
+		return $file;
+	}
+
+	$editor = wp_get_image_editor( $file['tmp_name'] );
+
+	if ( is_wp_error( $editor ) || ! $editor instanceof WP_Image_Editor_GD ) {
+		return $file;
+	}
+
+	// Check if required GD functions exist.
+	if (
+		! function_exists( 'imagecreatefrompng' ) ||
+		! function_exists( 'imageistruecolor' ) ||
+		! function_exists( 'imagealphablending' ) ||
+		! function_exists( 'imagesavealpha' ) ||
+		! function_exists( 'imagepalettetotruecolor' ) ||
+		! function_exists( 'imagepng' ) ||
+		! function_exists( 'imagedestroy' )
+	) {
+		return $file;
+	}
+
+	$image = imagecreatefrompng( $file['tmp_name'] );
+	if ( false === $image || imageistruecolor( $image ) ) {
```

Comment:
I'm confused by this check. If $image is false, the conversion failed. In that case, don't we want to continue to the processing below?